ecr/base.go

```diff
@@ -71,19 +65,26 @@ func (b *ecrBase) getManifest(ctx context.Context) (*ecr.Image, error) {
 		fmt.Println(err)
 		return nil, err
 	}
-	log.G(ctx).WithField("batchGetImage", batchGetImageOutput).Debug("ecr.base.manifest")
 
-	var ecrImage *ecr.Image
-	if len(batchGetImageOutput.Images) == 0 {
-		if len(batchGetImageOutput.Failures) > 0 &&
-			aws.StringValue(batchGetImageOutput.Failures[0].FailureCode) == ecr.ImageFailureCodeImageNotFound {
+	// Expect to recieve same number of images as requested.
+	if len(batchGetImageInput.ImageIds) == len(batchGetImageOutput.Images) {
+		return batchGetImageOutput.Images, nil
+	}
+
+	for _, failure := range batchGetImageOutput.Failures {
+		switch aws.StringValue(failure.FailureCode) {
+		// No matching image for combination of requested digest and image tag.
+		case ecr.ImageFailureCodeImageTagDoesNotMatchDigest:
+			return nil, errImageNotFound
```

Comment:
Yes, this let's matches be fully specified so that a `new tag || new manifest` are handled as entirely new combinations (causing the manifest to be sent with the new tag or as the tag's updated manifest).